### PR TITLE
Pepper_1458_Automate checking of External Path Review Dates

### PR DESCRIPTION
**Pull request by Rinchan20**
Handles this ticket: https://broadworkbench.atlassian.net/browse/PEPPER-1458 which is checking that the following fields:
Date Sent to External Path Review 
Date Received from External Path Review

are saving input as intended

Note: Test expanded to automate Tissue Request page for Tissue Type -> Block
Note x2: due to expansion, also handles this ticket (which is for automating the Tissue Type -> Block) as mentioned above: https://broadworkbench.atlassian.net/browse/PEPPER-1464

might need to add checking of the values' display in the Participant List to a different PR

### CODE REVIEW

**Comment by Rinchan20** [comment]
more checks needed but initial stuff is there (will add the check for OS2)

**Comment by Rinchan20** [comment]
just need to make a method to get back to the specific onc history I was modifying - check for presence of inputted info - download and check the download information

the slight expansion will basically automate the 'Tissue Block specific fields' regression test

**Comment by Rinchan20** [comment]
lms and os2 runs will check for the external path review dates (in addition to filling out usual fields for a tissue type -> block) - pancan only has the usual regular tissue request page checking (also tissue type -> block)

File: playwright-e2e/tests/dsm/tissue-request/cmi-tissue-request.spec.ts [hunk 1]
```diff
@@ -119,26 +153,235 @@ test.describe.skip('Tissue Request Flow', () => {
...
+          const customizeViewPanel = participantListPage.filters.customizeViewPanel;
+          await customizeViewPanel.open();
+          await customizeViewPanel.selectColumns(
+            'Tissue Columns',
+            [Label.DATE_SENT_FOR_EXTERNAL_PATH_REVIEW, Label.DATE_RECEIVED_FROM_EXTERNAL_PATH_REVIEW]
+          );
+          await customizeViewPanel.close();
+
+          //Download the Participant List
+          const download = await participantListPage.downloadParticipant({ fileFormat: FileFormat.XLSX });
```
Comment (Rinchan20) [inline playwright-e2e/tests/dsm/tissue-request/cmi-tissue-request.spec.ts]: this handy part is pretty much from the `download-all-data.spec.ts` test - modified very slightly for this test

**Comment by Rinchan20** [comment]
adding verification for the rest of the tissue request fields (for tissue type = 'block' for now) and checking them in participant list download soon

File: playwright-e2e/dsm/component/tissue.ts [hunk 2]
```diff
@@ -83,7 +85,7 @@ export default class Tissue extends ComponentBase {
     return value;
   }
 
-  public async fillSMIDs(SMID: SM_ID): Promise<SMID> {
+  public async getSMIDModal(SMID: SM_ID): Promise<SMID> {
```
Comment (Rinchan20) [inline playwright-e2e/dsm/component/tissue.ts:89]: renaming this so it's clearer what will happen when this method is called

File: playwright-e2e/dsm/pages/tablist-pages/tissue-request-page.ts [hunk 3]
```diff
@@ -101,12 +101,12 @@ export default class TissueRequestPage extends tablistPageBase {
...
     const destructionPolicyLocator = this.dynamicField(Label.DESTRUCTION_POLICY);
     const destructionPolicyYears = new Input(this.page, { root: destructionPolicyLocator });
-    const isInputDisabled = await destructionPolicyYears.isDisabled();
+    const isInputDisabled = await destructionPolicyYears.isDisabled('aria-checked');
 
     await this.checkCheckbox(destructionPolicyLocator, keptIndefinitelySelection);
 
     const existingValue = await destructionPolicyYears.currentValue();
-    const isAllowedToEnterValue = !keptIndefinitelySelection && !isInputDisabled && existingValue.trim() !== value.toString();
+    const isAllowedToEnterValue = (keptIndefinitelySelection === false && isInputDisabled === false && existingValue.trim() !== value.toString());
```
Comment (Rinchan20) [inline playwright-e2e/dsm/pages/tablist-pages/tissue-request-page.ts:109]: function was being weird all of the sudden so made a slight change

**Comment by Rinchan20** [comment]
large looking test but only seems to take ~ 1.7 minutes per study - is ran on pancan, lms, os2 - seems to have a total test run time of 5.5 minutes - a couple of things left to fix and validate

**Comment by Rinchan20** [comment]
Screenshot 2024-05-14 at 12 06 28 PM
verification of the smids in the participant list download doesn't mention which type of sm-id it is (e.g. USS, Scrolls, H&E) - but that seems to be following the trend seen in the download itself

File: playwright-e2e/tests/dsm/tissue-request/cmi-tissue-request.spec.ts [hunk 4]
```diff
@@ -524,119 +524,119 @@ test.describe('Tissue Request Flow', () => {
...
           logInfo(`Analyzing Materials Received for Scroll(s) -> Value: ${downloadedMaterialsReceivedForScrolls}`);
           expect(downloadedMaterialsReceivedForScrolls).toBe(testMaterialsReceivedValue);
 
-          const downloadedScrollsBackFromSHL = columnHeaderInRow['Scrolls back from SHL'].trim();
+          const downloadedScrollsBackFromSHL = valueFromColumn['Scrolls back from SHL'].trim();
           logInfo(`Analyzing Scrolls back from SHL -> Value: ${downloadedScrollsBackFromSHL}`);
           expect(downloadedScrollsBackFromSHL).toBe(todayInISOFormat);
 
-          const downloadedSHLWorkNumber = columnHeaderInRow['SHL Work Number'].trim();
+          const downloadedSHLWorkNumber = valueFromColumn['SHL Work Number'].trim();
```
Comment (zyme) [inline playwright-e2e/tests/dsm/tissue-request/cmi-tissue-request.spec.ts:576]: Nice naming, much easier to understand what value this is returning.

**Review by zyme — APPROVED** [review]

**Pull request merged**